src/Gateway/Surcharge.php

```diff
@@ -168,17 +234,34 @@ protected function addMaxLimit($fee, $gatewaySettings)
         return $fee;
     }
 
+    /**
+     * Return amount without the tax, so it can be included within the value
+     * @param float $amount
+     * @return float
+     */
+    protected function subtractTaxesFromValue(float $amount): float
+    {
+        if (!wc_tax_enabled() || $amount === 0.0) {
+            return $amount;
+        }
+        $taxRates = \WC_Tax::get_rates();
+        $wcStandardTax = \WC_Tax::calc_inclusive_tax($amount, $taxRates);
+        return $amount - $wcStandardTax[1];
+    }
+
     /**
      * @param string $gateway
-     *
      * @return string
      */
-    public function buildFeeName($gateway)
+    public function buildFeeName(string $gateway): string
     {
         return __($gateway, 'mollie-payments-for-woocommerce');
```

Comment:
btw this seems to be an invalid usage of `__`, you can translate only static strings.